### PR TITLE
Allow for commit status backref configuration

### PR DESCRIPTION
**Pull request by pupssman**
Via extension point

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/github-plugin/138)

### CODE REVIEW

**Comment by pupssman** [comment]
@lanwen plz C

**Comment by KostyaSha** [comment]
You need fill jenkins jira issue and reference it

**Comment by lanwen** [comment]
I think issue on Jira is not nessesary

File: src/main/java/org/jenkinsci/plugins/github/status/sources/BuildRefBackrefSource.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import hudson.model.TaskListener;
+
+/**
+ * Gets backref from Run URL.
+ *
+ * @author pupssman (Kalinin Ivan)
+ * @since 1.21.2
+ *
+ */
+public class BuildRefBackrefSource extends GitHubStatusBackrefSource {
```
Comment (oeuftete) [inline src/main/java/org/jenkinsci/plugins/github/status/sources/BuildRefBackrefSource.java]: https://issues.jenkins-ci.org/browse/JENKINS-38611

```text
Caused by: org.kohsuke.stapler.NoStaplerConstructorException: There's no @DataBoundConstructor on any constructor of class org.jenkinsci.plugins.github.status.sources.BuildRefBackrefSource
	at org.kohsuke.stapler.ClassDescriptor.loadConstructorParamNames(ClassDescriptor.java:247)
	at org.kohsuke.stapler.RequestImpl.instantiate(RequestImpl.java:758)
	at org.kohsuke.stapler.RequestImpl.access$200(RequestImpl.java:83)
	at org.kohsuke.stapler.RequestImpl$TypePair.convertJSON(RequestImpl.java:674)
	... 90 more
```

Comment (lanwen) [inline src/main/java/org/jenkinsci/plugins/github/status/sources/BuildRefBackrefSource.java]: will fix it in #141 in a few hours

**Pull request merged**